File: front/ChatBot/src/abstracts/message.ts
import {MessageOwner, MessageStatus, MessageType} from '../misc/message-type';

export default interface Message{
  id: string;
  content: string;
  type: MessageType;
  owner: MessageOwner;
  time: Date;
  status: MessageStatus;
  messageDialogId: string;
  sender?: string;
  questionId: string;
  question: string;
}

export interface MessageInfo{
  id: string;
  status: MessageStatus;
  messageDialogId: string;
}

File: back/src/Chatbot.Model/Data/Message.cs
using System;

namespace Chatbot.Model.Data
{
    public class Message
    {
        public Guid Id { get; set; }
        public string Content { get; set; }
        public MessageType Type { get; set; }
        public MessageOwner Owner { get; set; }
        public MessageStatus Status { get; set; }
        public DateTime Time { get; set; }
    }
}